lib/src/value/calculation.dart

```diff
@@ -158,6 +343,204 @@ class SassCalculation extends Value {
+        return result.unaryMinus();
+      }
+      return result.minus(dividend);
+    }
+    return result;
+  }
+
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if ((dividend is! SassNumber || modulus is! SassNumber) ||
+        !dividend.hasCompatibleUnits(modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    var strategyOrNumberSimplified = _simplify(strategyOrNumber);
+
+    switch ((
+      strategyOrNumberSimplified,
+      numberOrStep.andThen(_simplify),
+      step.andThen(_simplify)
+    )) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep('nearest', number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when {'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy.text, number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
```

Comment:
From the spec:

> If arguments has exactly two elements:
>
> If the first element is an unquoted string or interpolation with value "nearest", "up", "down", or "to-zero", and the second argument isn't an unquoted string or CalculationInterpolation, throw an error.

`SassNumber()` doesn't cover all the cases of arguments that aren't unquoted strings or interpolations. It's probably better to handle that case above here, and just allow any non-null value in second position in this case.

Reply:
Got it, I added the following case that returns a calculation: 
```
case ((SassString() || CalculationOperation() || SassNumber()) && var strategy, Object number, null)
```